### ISSUE TITLE
teach all humans to write code

### ISSUE THREAD

**Issue by dominictarr** [substack/omega-projects#11]
Once upon a time, only rich people could read and write. If you had proposed that _everyone_ should be literate, including women, black people etc, you would be laughed at. But now, in rich countries, very nearly everyone can read, and even in poor countries most people can read and in the worst situations... there is someone in the village who can read. Most importantly, the idea that everyone ought to be able to read is obvious and normal.

Not everyone can write a novel, or a poem, or reads books or even magazines, but they are able to read.

But, if you say "I think everyone should be able to write code" you'll get a similar response. "why would they want to?" "coding is too hard". I'm not saying everyone on earth should do a computer science degree. (it's not even that useful!) But everyone should be able to do a little bit...

It would obviously help if computers were easier to use - programming needs less edge cases (setting up compilers, etc, etc... code that breaks because dependencies have updated etc, configuring classpaths).

One approach to this would be: https://github.com/substack/omega-projects/issues/6

Once everyone could program then the concept of developer friendly and user friendly get closer.
It will also mean more people are able to contribute to open source... (which should contribute to feed open source developers / make *everything* open source)

**Comment by substack** [comment]
For programming knowledge to become more widespread, we'll need many more programmable devices than just computers. For some, computers are neat and interesting, but other endeavors are more interesting to other folks. I think a lot more fields will need to become programmable in concert with a broadening pool of programmers.

**Comment by dominictarr** [comment]
yeah, and even like, everyone has "smart" phones, but they are pretty dumb if you can't program on them.

**Comment by dominictarr** [comment]
at our current point though, there is benefit to having more people program in the normal way... and as programming becomes more common then the appeal of other products that are programmable would expand.

**Comment by mulderp** [comment]
Some related initiatives:
* Pamela Fox from Khan Academy: https://www.youtube.com/watch?v=aiFOurKwy7M
* Marvin Minsky about Why is Mathematics Hard to Learn: https://web.media.mit.edu/~minsky/OLPC-1.html
